### ISSUE TITLE
4k screen support

### ISSUE THREAD

**Issue by rkononov** [regolith-linux/regolith-desktop#28]
I've installed regolith on a laptop with 4k screen and UI is pretty small (fonts,status bar, system tray, indicators) - is there any way to set scale factor (like in gnome) - or maybe add one more section to README to explain how to tune regolith to hdpi screens?

**Comment by muniter** [comment]
Hello @rkononov If you want to scale your install, you will have to do it using the `Xft.dpi ` parameter on your `~/.Xresources` file. You can adjust it to your liking. Start setting values save the file, and then run `xrdb -merge ~/.Xresources` then reload i3, in this case the default binding is `$mod+Shift+r restart`, or you can just write on your terminal `i3-msg restart`

**Comment by kgilmer** [comment]
Thanks @muniter !  @rkononov another quick thing is to change the scaling in the Display control panel (Super-d).  I think you'll only be able to select 1x or 2x.  Hopefully @muniter 's suggestion will be more useful to you.  :smile:  Let us know if you are unable to get a scaling factor that works.

**Comment by rkononov** [comment]
hmm, I haven't found Xresources file on ubuntu (maybe Arch only), scaling in control panel helps with most of the apps but not i3 UI itself (panel, font size, drun). I'll check other options

**Comment by kgilmer** [comment]
I don't currently have access to a 4K monitor w/ high DPI @rkononov but I have in the past and do recall that updating the [DPI setting via xrandr](https://unix.stackexchange.com/a/274705/356932) did scale the bar, etc., but it could be that something has changed since then that's preventing it from working now.  I wonder if anyone else is using 4K w/ high DPI configurations and what they may have done to tune it...

**Comment by rkononov** [comment]
Fixed (partially) via increasing the font size in i3 config

**Comment by jzelner** [comment]
Hi All - this is perhaps a dumb question, but where does regolith call `exec i3`; I don't see an xinitrc file anywhere and the global Xsession doesn't reference i3 at all as far as I can tell? I would like to add these Xft.dpi changes to my startup if possible.

**Comment by kgilmer** [comment]
Here's the file you seek @jzelner : https://github.com/regolith-linux/regolith-gnome-flashback/blob/master/i3-gnome-flashback

Keep in mind that if you edit this file it may get overwritten by future package updates.   I believe that `apt` will prompt you but am not completely certain. 

Ideally there would be a way for you to add code to startup without having to modify these package scripts (due to the update issue).  I'm hoping there is a standard way of doing this but could also provide a hook in this script to an optional user script.

**Comment by jzelner** [comment]
Hi Ken,

Thanks! Is there any reason not to enable the new fractional gnome scaling
here, or will that not work in flashback? Also, is it necessary to change
the font DPI and also call xrandr or is Xft.dpi enough?

-Jon

**Comment by kgilmer** [comment]
It's my understanding that fractional scaling is a feature of a recent Gnome release that's beyond what's currently shipping in Ubuntu.  It's probably something we'll get in a subsequent release.  If you know of a way of getting that in Ubuntu 19.04 or 18.04 please let me know!

**Comment by shadowrylander** [comment]
Hello!
Quick question: to fix the status bar being so small, I edited my bar font size from 13 to 15; however, it is not applying on login / restart. I symlinked the modified regolith i3 config to `~/.config/i3/config` and `~/.i3/config`, but it's still seemingly loading the default, though it states:
```
Binary i3 version:  4.16.1-non-git © 2009 Michael Stapelberg and contributors          
Running i3 version: 4.16.1-non-git (pid 1382)c to abort…)                              
Loaded i3 config: /home/shadowrylander/.config/i3-regolith/config (Last modified: Wed 17
 Jul 2019 10:40:13 PM EDT, 7401 seconds ago)                                           

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```
where the associated regolith file is this:
```
# Configure the bar
bar {
  font pango:Source Code Pro Medium 15
  separator_symbol " "
  status_command i3blocks
  tray_output none

  colors {
      background $base03
      statusline $base1
      separator  $blue
#                        BORDER  BACKGROUND TEXT
      focused_workspace  $base02  $base02  $base2
      active_workspace   $base02  $base02  $base01
      inactive_workspace $base03  $base03  $base01
      urgent_workspace   $red     $red     $base3
  }
}
```
and it reloads every time I reload i3 using `mod - shift - r`.
How might I go about fixing this?

**Comment by apfsx** [comment]
@kgilmer Any word about this? Still no fractional scaling :+1:

**Comment by eddiewang** [comment]
same here, would love to see proper hidpi support with fractional scaling for multi-monitor displays.

**Comment by chrispitzer** [comment]
I've got a laptop with a high res screen and a big monitor that's normal resolution. In the standard ubuntu window manager I can go into displays and set my laptop to be at 200% scale, and the external monitor to be at 100% scale, and things mooooostly work well.

But in Regolith when I set one of them to 100% scale, both are set to 100% scale. Same for 200% scale. I cannot set them separately.

Is there a way that I can set the scale settings for my displays separately in Regolith? Is this a bug?
